### PR TITLE
CDD-2037: Working code

### PR DESCRIPTION
**Pull request by rashadmughal**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/notificationpushretry/connectors/CustomsNotificationConnector.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import scala.xml.XML
+
+class CustomsNotificationConnector @Inject()(config: ServiceConfiguration, http: HttpClient) {
+
+  private lazy val serviceBaseUrl: String = config.baseUrl("customs-notification")
+
+  def getNotifications()(implicit hc: HeaderCarrier): Future[BlockedCount] = {
+    http.GET[HttpResponse](s"$serviceBaseUrl/blocked-count").map(
+      response => {
+          BlockedCount((XML.loadString(response.body) \\ "pushNotificationBlockedCount").text.toInt)
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/customs/notificationpushretry/connectors/CustomsNotificationConnector.scala]: why parse the xml only to reconstruct it in the controller? Surely it's far more efficient to pass the body of the response back to the controller and pass it on to the client untouched?

File: app/uk/gov/hmrc/customs/notificationpushretry/controllers/NotificationsController.scala [hunk 2]
```diff
@@ -18,12 +18,33 @@ package uk.gov.hmrc.customs.notificationpushretry.controllers
...
-class NotificationsController @Inject()() extends BaseController {
+class NotificationsController @Inject()(customsNotificationService: CustomsNotificationService,
+                                        headerValidator: HeaderValidator,
+                                        logger: CdsLogger) extends BaseController {
+
+  def get(): Action[AnyContent] =
+    (headerValidator.validateAcceptHeader andThen headerValidator.validateXClientIdHeader).async { implicit request =>
+
+      def xmlResult(pushNotificationBlockedCount: Int) = <pushNotificationBlockedCount>{pushNotificationBlockedCount}</pushNotificationBlockedCount>
+
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/notificationpushretry/controllers/NotificationsController.scala]: We should not propagate HeaderCarrier to connector, connector should create its own HeaderCarrier

File: test/unit/controllers/NotificationsControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import uk.gov.hmrc.customs.api.common.logging.CdsLogger
+import uk.gov.hmrc.customs.notificationpushretry.controllers.NotificationsController
+import uk.gov.hmrc.customs.notificationpushretry.model.BlockedCount
+import uk.gov.hmrc.customs.notificationpushretry.services.CustomsNotificationService
+import uk.gov.hmrc.customs.notificationpushretry.validators.HeaderValidator
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.Future
+
```
Comment (googley42) [inline test/unit/controllers/NotificationsControllerSpec.scala]: how about a test for 404?

File: test/component/ExternalServices.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (googley42) [inline test/component/ExternalServices.scala]: no logging as `DEBUG` or `INFO` (eg for header validation) that includes ClientId

File: app/uk/gov/hmrc/customs/notificationpushretry/connectors/CustomsNotificationConnector.scala (hunk 1 above)
Comment (googley42) [reply app/uk/gov/hmrc/customs/notificationpushretry/connectors/CustomsNotificationConnector.scala]: strictly speaking each microservice should control its own output, but I guess in this case it is so simple we can pass through

File: build.sbt [hunk 5]
```diff
@@ -134,3 +134,21 @@ unmanagedResourceDirectories in Compile += baseDirectory.value / "public"
 
 libraryDependencies ++= compileDependencies ++ testDependencies
```
Comment (googley42) [inline build.sbt]: comment / code should not mention WCO

**Pull request merged**